packages/storefront/src/components/PropsTable.vue

```diff
@@ -0,0 +1,155 @@
+          <td>
+            <code>{{
+              (typeof value.defaultValue === 'string' ? `'${value.defaultValue}'` : value.defaultValue) ?? 'undefined'
+            }}</code>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import Component from 'vue-class-component';
+  import { Prop } from 'vue-property-decorator';
+  import { paramCase } from 'change-case';
+  import { getComponentMeta, type ComponentMeta, type PropMeta } from '@porsche-design-system/component-meta';
+  import type { TagName } from '@porsche-design-system/shared';
+
+  const formatHtml = (input: string): string =>
+    input.replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/\n/g, '<br>');
+
+  const wrapInCodeCode = (input: string): string => `<code>${formatHtml(input)}</code>`;
+
+  @Component
+  export default class PropsTable extends Vue {
+    @Prop() public component!: TagName;
+
+    paramCase = paramCase;
+    breakpointCustomizableGeneric = `type BreakpointCustomizable<T> = {
+  base: T;
+  xs?: T;
+  s?: T;
+  m?: T;
+  l?: T;
+  xl?: T;
+  xxl?: T;
+}`;
+
+    public get propsMeta(): ComponentMeta['propsMeta'] {
+      const unorderedPropsMeta = getComponentMeta(this.component)?.propsMeta;
+      return unorderedPropsMeta
+        ? Object.keys(unorderedPropsMeta)
+            .sort()
+            .reduce((result, key) => ({ ...result, [key]: unorderedPropsMeta[key] }), {})
+        : {};
+    }
+
+    public get hasBreakpointCustomizableProp(): boolean {
+      return this.propsMeta
+        ? Object.values(this.propsMeta).some((propMeta) => propMeta.isBreakpointCustomizable)
+        : false;
+    }
+
+    public formatDescription(meta: PropMeta): string {
+      return meta.description
+        ? meta.description
+            .replace(/@(deprecated)/, '<strong class="deprecated">$1</strong>') // deprecated annotation
+            .replace(/`(.+?)`/g, (_, g1) => `<code>${formatHtml(g1)}</code>`) // prop references in backticks
+        : '';
+    }
+
+    public formatType(meta: PropMeta): string {
+      return Array.isArray(meta.allowedValues) || meta.isBreakpointCustomizable
+        ? [
+            ...(meta.type.includes('|')
+              ? [] // inline union type is taken care of via allowedValues
+              : [
+                  meta.type !== 'string' && meta.type !== 'number' && meta.type !== 'boolean'
+                    ? `type ${meta.type} =` // literal types, etc.
+                    : meta.type, // simple type
+                ]),
+            ...(Array.isArray(meta.allowedValues)
+              ? meta.allowedValues.map(
+                  (val) =>
+                    wrapInCodeCode(
+                      val === 'string'
+                        ? val
+                        : val === null
+                          ? 'undefined'
+                          : typeof val === 'number'
+                            ? `${val}`
```

Comment:
Do we really want to write code with all of these inline conditions and shorthand syntax, seams hard to debug later on?

Reply:
Refactored it to `formatPropType()` util with unit tests.